conans/client/graph/graph_binaries.py

```diff
@@ -343,6 +343,16 @@ def evaluate_graph(self, deps_graph, build_mode, update, remotes, nodes_subset=N
         for node in deps_graph.ordered_iterate(nodes_subset=nodes_subset):
             self._propagate_options(node)
 
+            # Make sure that locked options match
+            if (node.graph_lock_node is not None and
```

Comment:
This has replaced the previous ``options.freeze()`` instead of detecting when someone is trying to modify a frozen option value, check at the end of the process that the final options match the locked ones.